### PR TITLE
Fixed complaints for CA1825

### PR DESCRIPTION
**Pull request by Exouxas**
Modified all occurrences of zero-length array allocations to use Array.Empty<datatype>(). 

Addresses #791

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=794) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=794) before we can accept your contribution.<br/><br/>:x: Exouxas [sign now](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=794)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/SteeltoeOSS/Steeltoe?pullRequest=794) it.</sub>

File: src/Messaging/test/MessagingBase.Test/Handler/DestinationPatternsMessageConditionTest.cs [hunk 1]
```diff
@@ -132,7 +132,7 @@ private DestinationPatternsMessageCondition Condition(params string[] patterns)
 
         private IMessage MessageTo(string destination)
         {
-            return MessageBuilder.WithPayload(new byte[0]).SetHeader(
+            return MessageBuilder.WithPayload(Array.Empty<byte>()).SetHeader(
```
Comment (TimHess) [inline src/Messaging/test/MessagingBase.Test/Handler/DestinationPatternsMessageConditionTest.cs:135]: Please add `using System;` to this file and then the PR should be ready to merge, thank you!

**Review by TimHess — CHANGES_REQUESTED** [review]

**Comment by Exouxas** [comment]
@TimHess I've added `using System;` now, but I'm a bit confused by the resulting commits. Can you verify that it's correct?

Tried to reset and kept changes so it would all be in one commit, but it looks like it created 2 commits and 1 merge instead..

**Comment by TimHess** [comment]
> @TimHess I've added `using System;` now, but I'm a bit confused by the resulting commits. Can you verify that it's correct?
> 
> Tried to reset and kept changes so it would all be in one commit, but it looks like it created 2 commits and 1 merge instead..

Sorry about the confusion there - I had merged another (important, pipeline-related) commit in main and then pushed the update button here.. I'll squash during the merge so it'll only be 1 commit in the end and nothing to worry about

**Review by TimHess — APPROVED** [review]
LGTM, thanks!

**Pull request merged**